### PR TITLE
Support request body matching (WireMock bodyPatterns)

### PR DESCRIPTION
**Pull request by adamtheturtle**
Closes #228.

Adds support for matching requests on their body via WireMock `bodyPatterns`, so two requests to the same method and URL can return different responses based on their bodies. Supported matchers are `equalToJson` (honouring `ignoreArrayOrder` and `ignoreExtraElements`, accepting either a JSON value or a JSON string), `contains`, and `equalTo`, implemented as a custom respx `Pattern` with no new runtime dependencies. The change includes 26 new tests at 100% coverage, README and changelog updates, and a `uv.lock`/`pyproject.toml` sync that the repo's own pre-commit hooks required. `matchesJsonPath` is deliberately deferred because it needs an untyped JSONPath engine that clashes with this repo's strict type-checking config.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> New request-routing behavior is non-trivial JSON matching and unsupported `bodyPatterns` types are skipped rather than failing, so WireMock parity edge cases could surprise tests until exercised.
> 
> **Overview**
> Adds **WireMock `bodyPatterns`** so stubs can distinguish requests that share method and URL by body, wiring matchers into **respx** via a custom `_BodyPattern` on each route.
> 
> Supported matchers are **`equalToJson`** (with `ignoreArrayOrder` / `ignoreExtraElements`, object or JSON string expectations), **`contains`**, and **`equalTo`**. Multiple patterns on one stub must all match. JSON comparison is implemented in-library (no new runtime deps). README, a newsfragment, and a large pytest suite cover the behavior; dev-tool versions in `uv.lock` / `pyproject.toml` were bumped as part of the same change.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 496e1f70365a717eda954fc7a1bda6b121c1c5f8. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/wiremock_mock/__init__.py [hunk 1]
```diff
@@ -32,6 +36,269 @@ class _ResponseSpec(TypedDict, total=False):
...
+    """Build a predicate matching the request body as JSON."""
+
+    def predicate(request: httpx.Request) -> bool:
+        """Return whether the request body matches ``expected`` as
+        JSON.
+        """
+        try:
+            actual = json.loads(s=request.read())
+        except json.JSONDecodeError:
+            return False
```
[the remark this answers is not in the document]
Comment (adamtheturtle) [reply src/wiremock_mock/__init__.py:230]: Fixed in fc985050. `_equal_to_json_predicate` now decodes the body via `_request_text` and treats a non-UTF-8 body as a non-match (returns `False`), matching `contains`/`equalTo` so route matching is never aborted. Added a regression test (`test_add_wiremock_to_respx_body_equal_to_json_non_utf8`).
